0024-pre-shared-key-v2/0024-pre-shared-key-v2.md

```diff
@@ -0,0 +1,266 @@
+7. Receiver rejects incoming transfer, setting the rejection message's data to the encrypted quote response.
+8. Sender decrypts the quote response and divides the receiver's Chunk Amount by the source amount to determine the path exchange rate.
+
+### Single Payment
+
+1. In the case of a fixed source amount payment, sender multiplies the source amount by the rate to determine the Payment Amount. In the case of a fixed destination amount payment, sender divides the destination amount by the rate to determine the amount for their outgoing transfer.
+2. Sender creates a packet with a Type of 1, a random Payment ID, a Sequence number of 0, and the Payment Amount and Chunk Amount set to the destination amount determined in step 2. Sender [encrypts](#encryption-envelope) the request.
+3. Sender generates the payment condition as described [below](#fulfillable-condition).
+4. Sender sends a payment using the source amount determined in step 2, the condition from step 3, and the encrypted request as the data.
+5. Receiver is notified of the incoming transfer and decrypts the data.
+6. Receiver checks to ensure that the incoming transfer amount is greater than or equal to the Chunk Amount specified in the request. If the Payment Amount is not equal to 0, the receiver checks that the transfer amount is less than or equal to the Payment Amount specified in the request.
+7. Receiver creates a [Payment Chunk Response](#payment-chunk-response) with the same Payment ID and Sequence number. The Payment Amount and Chunk Amount are set to the amount that arrived on the incoming transfer. The receiver then [encrypts](#encryption-envelope) the data.
+8. Receiver generates fulfillment as described [below](#fulfillment-generation) and fulfills the incoming transfer with the fulfillment and the encrypted response data.
+9. Sender is notified of the outgoing transfer being fulfilled and can use the decrypted response to determine the amount that arrived at the destination.
+
+### Chunked Payment
+
+1. Sender creates a packet with Type 0, a random Payment ID, a Sequence number of 0 and a Chunk Amount of 0. In the case of a fixed destination amount, sender sets the Payment Amount to the destination amount. In the case of a fixed source amount, sender sets the Payment Amount to 18446744073709551615 (the maximum possible value). Sender [encrypts the request](#encryption-envelope).
+2. Sender generates the payment condition as described [below](#fulfillable-condition).
+3. Sender sends a payment with the condition from step 2., the encrypted request from step 1. as the data, and an arbitrary source amount.
+4. Receiver is notified of the incoming transfer and decrypts the data.
+5. Receiver checks that the incoming transfer amount is greater than or equal to the Chunk Amount and that the total that has arrived thus far does not exceed the Payment Amount.
```

Comment:
At step 1) a packet with Chunk Amount of 0 is created. So how can this constraint ever be violated:
> Receiver checks that the incoming transfer amount is greater than or equal to the Chunk Amount [...]

 The incoming transfer would need to have an amount < 0 to violate this constraint.

Reply:
I assume for consecutive packets the chunk amount would be greater than 0.